modules/es-extensions/publisher/extensions/app/greg-apis/apis/permission.jag

```diff
@@ -0,0 +1,148 @@
+        return rxt.asset.createUserAssetManager(session, type);
+    };
+
+    var checkPermission = function (assetType, assetId) {
+        var am = assetManager(session, assetType);
+        var rolePermission = gregAPI.permissions.list(am, assetId);
+
+        if (rolePermission) {
+            return (rolePermission.isAuthorizeAllowed && !rolePermission.isVersionView);
+        } else {
+            return false;
+        }
+    };
+
+    switch (httpMethod) {
+        case 'POST':
+            assetType = params.assetType;
+            assetId = params.assetId;
+            am = assetManager(session, assetType);
+
+            if (checkPermission(assetType, assetId)) {
+                try {
+                    results = gregPermissionUtil.permissions.add(am, params);
+                    message = {'status': results};
+                    print(message);
+                } catch (e) {
+                    log.error(e);
+                    message = {'error': e};
+                    print(message);
+                }
+
+            } else {
+                message = {'status': false};
+                response.status = 401;
+                print(message);
+            }
+            break;
+        case 'GET':
+            assetType = options.assetType;
+            assetId = options.id;
+
+            if (assetId) {
+                am = assetManager(session, assetType);
+                if (checkPermission(assetType, assetId)) {
+                    try {
+                        results = gregPermissionUtil.permissions.list(am, assetId);
+                        if (results != null) {
+                            message = {'data': results};
+                        } else {
+                            message = {'data': {}};
+                            response.status = 401;
+                        }
+                        print(message);
+                    } catch (e) {
+                        log.error(e);
+                        message = {'error': e};
+                        print(message);
+                    }
+                } else {
+                    message = {'data': {'status': false}};
+                    response.status = 401;
+                    print(message);
+                }
+            }
+            break;
+        case 'DELETE':
+            assetType = options.assetType;
+            assetId = options.assetId;
+            am = assetManager(session, assetType);
+
+            if (checkPermission(assetType, assetId)) {
+                try {
+                    results = gregPermissionUtil.permissions.remove(am, options);
+                    message = {'status': results};
+                    print(message);
+                } catch (e) {
+                    log.error(e);
+                    message = {'error': e};
+                    print(message);
+                }
+            } else {
+                message = {'status': false};
+                response.status = 401;
```

Comment:
Thanks for commenting. Fixed with the following PR - https://github.com/wso2/product-greg/pull/851